TSRM/tsrm_win32.c

```diff
@@ -619,8 +619,8 @@ TSRM_API int shmget(key_t key, size_t size, int flags)
 	snprintf(shm_segment, sizeof(shm_segment), "TSRM_SHM_SEGMENT:%d", key);
 	snprintf(shm_info, sizeof(shm_info), "TSRM_SHM_DESCRIPTOR:%d", key);
 
-	shm_handle  = OpenFileMapping(FILE_MAP_ALL_ACCESS, FALSE, shm_segment);
-	info_handle = OpenFileMapping(FILE_MAP_ALL_ACCESS, FALSE, shm_info);
+	shm_handle  = OpenFileMapping(FILE_MAP_ALL_ACCESS, FALSE, key == IPC_PRIVATE ? NULL : shm_segment);
+	info_handle = OpenFileMapping(FILE_MAP_ALL_ACCESS, FALSE, key == IPC_PRIVATE ? NULL : shm_info);
```

Comment:
shm_segment/shm_info are also used below in CreateFileMapping.

Reply:
Bad PR, sorry!